MultisigTests/Models/SafeTests.swift

```diff
@@ -10,10 +10,37 @@ import XCTest
 @testable import Multisig
 
 class SafeTests: CoreDataTestCase {
+
+    func test_removeSafe() throws {
+        Safe.create(address: "0", name: "0")
+        Safe.create(address: "1", name: "1", selected: true)
+        Safe.create(address: "2", name: "2")
+
+        let context = CoreDataStack.shared.viewContext
+        var result = try context.fetch(Safe.fetchRequest().all())
+        XCTAssertEqual(result.count, 3)
+
+        var safe = result.first!
+        Safe.remove(safe: safe)
+        result = try context.fetch(Safe.fetchRequest().all())
+        XCTAssertEqual(result.count, 2)
+
```

Comment:
I would also check that the selected safe is the first one (result.first?.isSelected == true)

same for other checks;

and at the end, the Selection must be niil